module/metrics/noop.go

```diff
@@ -234,6 +234,7 @@ func (nc *NoopCollector) PrunedBlocks(totalByHeight, totalById, storedByHeight,
 func (nc *NoopCollector) RangeRequested(ran chainsync.Range)                                    {}
 func (nc *NoopCollector) BatchRequested(batch chainsync.Batch)                                  {}
 func (nc *NoopCollector) OnUnauthorizedMessage(role, msgType, topic, offense string)            {}
+func (nc *NoopCollector) MetricsMiddleware()                                                    {}
```

Comment:
These would be the metrics methods (e.g. `ObserveHTTPRequestDuration`). The idea is that in tests you can instantiate a `NoopCollector` and pass it into any method that expects a metrics object.